### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by KonradStaniec**
Pipeline metric by pushing to gateway

### CODE REVIEW

File: internal/services/unbonding_pipeline.go [hunk 1]
```diff
@@ -213,6 +226,22 @@ func outputsAreEqual(a, b *wire.TxOut) bool {
...
+func (up *UnbondingPipeline) pushMetrics() error {
+	gateWayAdderss, err := up.Metrics.Config.Address()
+	if err != nil {
+		return fmt.Errorf("failed to get gateway address: %w", err)
+	}
+
+	up.logger.Info("Pushing metrics to gateway", "gateway", gateWayAdderss)
+
+	return push.New(gateWayAdderss, "unbonding-pipeline").
+		Collector(up.Metrics.SuccessigningReqs).
```
Comment (vitsalis) [inline internal/services/unbonding_pipeline.go:238]:
```suggestion
		Collector(up.Metrics.SuccessSigningReqs).
```

File: internal/services/unbonding_pipeline.go [hunk 2]
```diff
@@ -213,6 +226,22 @@ func outputsAreEqual(a, b *wire.TxOut) bool {
...
+		Collector(up.Metrics.FailedsigningReqs).
```
Comment (vitsalis) [inline internal/services/unbonding_pipeline.go:239]:
```suggestion
		Collector(up.Metrics.FailedSigningReqs).
```

File: internal/services/unbonding_pipeline_metrics.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package services
+
+import (
+	"github.com/babylonchain/cli-tools/internal/config"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type PipelineMetrics struct {
+	SuccessigningReqs          *prometheus.CounterVec
```
Comment (vitsalis) [inline internal/services/unbonding_pipeline_metrics.go:9]:
```suggestion
	SuccessSigningReqs          *prometheus.CounterVec
```

File: internal/services/unbonding_pipeline_metrics.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	FailedsigningReqs          *prometheus.CounterVec
```
Comment (vitsalis) [inline internal/services/unbonding_pipeline_metrics.go:10]:
```suggestion
	FailedSigningReqs          *prometheus.CounterVec
```

File: internal/services/unbonding_pipeline.go [hunk 5]
```diff
@@ -10,6 +10,9 @@ import (
 	"github.com/babylonchain/cli-tools/internal/btcclient"
 	"github.com/babylonchain/cli-tools/internal/config"
 	"github.com/babylonchain/cli-tools/internal/db"
+	"github.com/prometheus/client_golang/prometheus/push"
+
+	// "github.com/prometheus/client_golang/prometheus/promauto"
```
Comment (vitsalis) [inline internal/services/unbonding_pipeline.go:15]: leftover?

**Review by vitsalis — APPROVED** [review]

**Comment by KonradStaniec** [comment]
Hey @filippos47 lets tests commit https://github.com/babylonchain/cli-tools/pull/33/commits/2d06738bc0f1d04c44b8ec675fb65c0311c5b1da, with prometheus gateway before merging this pr

**Comment by hiepmai-babylonchain** [comment]
Tested locally, metrics are sent to pushgateway

**Review by hiepmai-babylonchain — APPROVED** [review]

**Pull request merged**